builder/azure/arm/template_factory.go

```diff
@@ -51,6 +51,13 @@ func GetVirtualMachineDeployment(config *Config) (*resources.Deployment, error)
 		builder.SetMarketPlaceImage(config.ImagePublisher, config.ImageOffer, config.ImageSku, config.ImageVersion)
 	}
 
+	if config.VirtualNetworkName != "" {
+		builder.SetVirtualNetwork(
+			config.VirtualNetworkName,
```

Comment:
I've noticed an error on deployment - I've traced it back to here. 

```
Resource /subscriptions/XXX/resourceGroups/VNET-TOM/providers/Microsoft.Network/virtualNetworks/CORE-DEVELOP referenced by resource /subscriptions/XXX/resourceGroups/packer-Resource-Group-q3xwzvkzwz/providers/Microsoft.Network/networkInterfaces/packerNic was not found. Please make sure that the referenced resource exists, and that both resources are in the same region. (Code: InvalidResourceReference)
```

The problem is the order of the parameters passed to the SetVirtualNetwork method, the order should be as follows:

```
builder.SetVirtualNetwork(
    config.VirtualNetworkResourceGroupName,
    config.VirtualNetworkName,
    config.VirtualNetworkSubnetName)
```